src/tiledb/cloud/geospatial/ingestion.py

```diff
@@ -0,0 +1,1182 @@
+                    "chunk_size": chunk_size,
+                }
+            )
+            samples = meta.paths[1:]
+        else:
+            raise ValueError("Require at least one point cloud file to have been found")
+    elif dataset_type == "raster":
+        fn = ingest_raster_udf
+
+        if search_uri:
+            if pattern:
+                sources = find(
+                    search_uri,
+                    config=config,
+                    excludes=ignore,
+                    includes=pattern,
+                    max_files=max_files,
+                )
+            else:
+                sources = find(
+                    search_uri,
+                    config=config,
+                    excludes=ignore,
+                    includes=raster_match,
+                    max_files=max_files,
+                )
+
+        if len(sources) == 0:
+            raise ValueError("No raster datasets found")
+
+        meta = get_raster_metadata(
+            sources, config=config, dst_tile_size=tile_size, res=res
+        )
+        kwargs.update(
+            {
+                "crs": meta.crs,
+                "extents": meta.extents,
+                "res": meta.res,
+                "band_count": meta.band_count,
+                "dtype": meta.dtype,
+                "nodata": nodata,
+                "resampling": resampling,
+                "tile_size": tile_size,
+                "compressor": compression_filter,
+            }
+        )
+        samples = meta.block_metadata
+    elif dataset_type == "geometry":
+        fn = ingest_geometry_udf
+
+        if search_uri:
+            if pattern:
+                sources = find(
+                    search_uri,
+                    config=config,
+                    excludes=ignore,
+                    includes=pattern,
+                    max_files=max_files,
+                )
+            else:
+                sources = find(
+                    search_uri,
+                    config=config,
+                    exclude=ignore,
+                    includes=geometry_match,
+                    max_files=max_files,
+                )
+
+        full_extents = get_geometry_metadata(sources, config=config)
+        kwargs.update(
+            {
+                "extents": full_extents,
+            }
+        )
+    else:
+        raise ValueError(f"Unsupported ingestion dataset type: {dataset_type}")
+
+    if not unit_testing:
+        # Build the task graph
+        dag_name = f"{dataset_type}-{DEFAULT_DAG_NAME}"
+        task_prefix = f"{dataset_type} - Task"
+
+        logger.info("Building graph")
+        graph = tiledb.cloud.dag.DAG(
+            name=dag_name,
+            mode=tiledb.cloud.dag.Mode.BATCH,
+            max_workers=workers,
+            namespace=namespace,
+        )
+
+        nodes = []
+
+        # create first node
+        ingest_node = graph.submit(
+            fn,
+            dataset_uri=dataset_uri,
+            config=config,
```

Comment:
What is the purpose of this node? how does it take different arguments but calling the same `fn` with the rest below?

Reply:
This node is the set up node and doesn't take any source files just the metadata needed to construct an empty array.